modules/mcuboot/hooks/nrf53_hooks.c

```diff
@@ -19,9 +19,51 @@
 #define NET_CORE_VIRTUAL_PRIMARY_SLOT 3
 
 #include <dfu/pcd.h>
+#include <fw_info_bare.h>
 
-int boot_read_image_header_hook(int img_index, int slot,
-		struct image_header *img_head)
+/**
+ * Compare image version numbers
+ * @retval -1           If ver1 is less than ver2.
+ * @retval 0            If the image version numbers are equal.
+ * @retval 1            If ver1 is greater than ver2.
+ */
+int boot_version_cmp_net(const struct flash_area *fap, struct image_header *hdr)
+{
+	int err;
+	const struct fw_info *firmware_info;
+	uint32_t version = 0;
+	uint32_t read_flash_page[4096];
+
+	err = pcd_network_core_app_version(&version);
+	if (err != 0) {
+		BOOT_LOG_ERR("Failure fetching network core application version");
+		return -EFAULT;
+	}
+
+	err = flash_area_read(fap, hdr->ih_load_addr, &read_flash_page, 4096 * sizeof(uint32_t));
```

Comment:
we need to use a define for 4096 here, why not ARRAY_SIZE directly?

Reply:
Forgot about the array_size macro